### PR TITLE
[M2P-504] Link order with newly created customer during sso login

### PR DESCRIPTION
**Pull request by adentong**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: (link Jira ticket)

#changelog [M2P-504] Link order with newly created customer during sso login

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Model/Api/OAuthRedirect.php [hunk 1]
```diff
@@ -221,9 +241,22 @@ public function login($code = '', $scope = '', $state = '')
...
+                if ($order !== false && !$order->getCustomerId() && $order->getBillingAddress()->getEmail() === $payload['email']) {
+                    $order->setCustomerId($customer->getId());
+                    $order->setCustomerIsGuest(0);
+                    $this->orderRepository->save($order);
+                }
+            }
+
             $this->linkLoginAndRedirect($externalID, $customer->getId());
         } catch (Exception $e) {
+            $this->bugsnag->notifyError('OAuthRedirect', $e->getMessage());
```
Comment (vitaliyreznikov) [inline Model/Api/OAuthRedirect.php:259]: It's better to send all exception object into bugsnag, not only message. In this case, we will know the exact line where error happened.

**Review by vitaliyreznikov — APPROVED** [review]
LGTM, one minor comment

File: Model/Api/OAuthRedirect.php (hunk 1 above)
Comment (adentong) [reply Model/Api/OAuthRedirect.php:259]: will update

**Review by njfaries — APPROVED** [review]

File: Model/Api/OAuthRedirect.php (hunk 1 above)
Comment (adentong) [reply Model/Api/OAuthRedirect.php]: updated.

**Review by njfaries — APPROVED** [review]

**Review by nickfiacco-bolt — APPROVED** [review]

**Pull request merged**